Tests/FluentAssertions.Specs/Types/MethodInfoSelectorAssertionSpecs.cs

```diff
@@ -231,5 +232,101 @@ public void When_asserting_methods_are_not_decorated_with_attribute_but_they_are
                              "*ClassWithAllMethodsDecoratedWithDummyAttribute.ProtectedDoNothing*" +
                              "*ClassWithAllMethodsDecoratedWithDummyAttribute.PrivateDoNothing");
         }
+
+
+
+        [Fact]
+        public void When_asserting_methods_have_specified_accessor_it_should_succeed()
+        {
+            // Arrange
+            var methodSelector = new MethodInfoSelector(typeof(ClassWithPublicMethods));
+
+            // Act
+            Action act = () =>
+                methodSelector.Should().Be(CSharpAccessModifier.Public);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_asserting_methods_does_not_have_specified_accessor_it_should_throw()
```

Comment:
I would rephrase tests like this `When_not_all_methods_have_the_specified_access_modifier_it_should_throw`. It avoids the repetitive `When_asserting_methods` and keeps them functional.

Reply:
Fixed